modality-probe-cli/src/manifest_gen/rust_parser.rs

```diff
@@ -330,28 +370,40 @@ fn event_call_exp(input: Span) -> ParserResult<Span, EventMetadata> {
         .map_err(|e| convert_error(e, Error::MissingSemicolon(pos.into())))?;
     let (input, _) =
         tag(");")(input).map_err(|e| convert_error(e, Error::MissingSemicolon(pos.into())))?;
-    let mut arg_vec: Vec<&str> = args
-        .fragment()
-        .split(',')
-        .map(|s| s.trim())
-        .filter(|s| !s.is_empty())
-        .collect();
-    match arg_vec.len() {
-        2..=4 => (), // Probe and event name required, maybe tags and description
-        _ => return Err(make_failure(input, Error::Syntax(pos.into()))),
-    }
-    let arg0 = Span::new_extra(arg_vec.remove(0), input.extra);
-    let (_, agent_instance) =
-        rest_string(arg0).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?;
-    let arg1 = Span::new_extra(arg_vec.remove(0), input.extra);
-    let (_, name) = alt((reduced_event_id_exp_alt_a, reduced_event_id_exp_alt_b))(arg1)
-        .map_err(|e| convert_error(e, Error::Syntax(pos.into())))?;
+    let (args, agent_instance) =
+        variable_call_exp_arg(args).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?;
+    let expect_tags_or_desc = peek(variable_call_exp_arg)(args).is_ok();
+    let (args, full_name) = if expect_tags_or_desc {
+        variable_call_exp_arg(args).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?
+    } else {
+        let (args, name_token) =
+            take_until(")")(args).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?;
+        let (_args, _) = tag(")")(args).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?;
+        rest_string(name_token).map_err(|e| convert_error(e, Error::Syntax(pos.into())))?
+    };
+    let arg = Span::new_extra(&full_name, input.extra);
+    let (_, name) = alt((reduced_event_id_exp_alt_a, reduced_event_id_exp_alt_b))(arg)
+        .map_err(|_| make_failure(input, Error::Syntax(pos.into())))?;
     let name =
         reduce_namespace(&name).map_err(|_| make_failure(input, Error::Syntax(pos.into())))?;
     if !event_name_valid(&name) {
         return Err(make_failure(input, Error::Syntax(pos.into())));
     }
-    let mut tags_and_desc: Vec<String> = arg_vec.iter().map(|s| (*s).to_string()).collect();
+    let mut arg_vec: Vec<String> = Vec::new();
+    let mut iter = iterator(args, multi_variable_call_exp_arg_literal);
+    //iter.for_each(|s| arg_vec.push(s));
+    // TODO
```

Comment:
What's this TODO and the commented out line above about?

Reply:
Oops, accidental, will get that fixed up.